core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java

```diff
@@ -121,7 +121,7 @@ public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long
       }
       throw new IOException(e);
     } finally {
-      if (channel != null && listener != null) {
+      if (channel != null && listener != null && channel.isActive()) {
```

Comment:
i can throw an exception (NullPointerException) explicitly. Is that what you are suggesting?. It should not be null in the current code. If it happens to null, something is working unexpectedly.